packages/faucet/src/api/requestparser.ts

```diff
@@ -45,15 +50,39 @@ export class RequestParser {
     if (denom.length === 0) {
       throw new HttpError(400, "Property 'denom' must not be empty.");
     }
+    
+    if (typeof email !== "string") {
+      throw new HttpError(400, "Property 'email' must be a string.");
+    }
+
+    if (!email.match(/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/)) {
```

Comment:
This will likely fail on some email addresses, such as example@cheqd.foundation or on user+faucet@cheqd.io.

Use a more comprehensive regex, or consider using an email validation package: https://www.golinuxcloud.com/javascript-validate-email/